src/main/java/cws/k8s/scheduler/scheduler/NodeLabelAssign.java

```diff
@@ -0,0 +1,97 @@
+import cws.k8s.scheduler.client.KubernetesClient;
+import cws.k8s.scheduler.scheduler.nodeassign.LabelAssign;
+import cws.k8s.scheduler.scheduler.nodeassign.NodeAssign;
+import cws.k8s.scheduler.scheduler.nodeassign.FairAssign;
+import cws.k8s.scheduler.util.NodeTaskAlignment;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class NodeLabelAssign extends Scheduler {
+
+    private final Prioritize prioritize;
+    private final NodeAssign nodeAssigner;
+    private final NodeAssign nodeLabelAssigner;
+
+    public NodeLabelAssign(String execution, KubernetesClient client, String namespace, SchedulerConfig config) {
+        this(execution, client, namespace, config, new RankMaxPrioritize(), new LabelAssign(config), new FairAssign());
+    }
+
+    public NodeLabelAssign( String execution,
+                                      KubernetesClient client,
+                                      String namespace,
+                                      SchedulerConfig config,
+                                      Prioritize prioritize,
+                                      NodeAssign nodeLabelAssigner,
+                                      NodeAssign nodeAssigner ) {
+        super(execution, client, namespace, config);
+        this.prioritize = (prioritize != null) ? prioritize : new RankMaxPrioritize();
+        this.nodeLabelAssigner = (nodeLabelAssigner != null) ? nodeLabelAssigner : new LabelAssign(config);
+        this.nodeAssigner = (nodeAssigner != null) ? nodeAssigner : new FairAssign();
+        nodeAssigner.registerScheduler( this );
+        if ( nodeAssigner instanceof Informable ){
+            client.addInformable( (Informable) nodeAssigner );
+        }
+    }
+
+    @Override
+    public void close() {
+        super.close();
+        if ( nodeAssigner instanceof Informable ){
+            client.removeInformable( (Informable) nodeAssigner );
+        }
+    }
+
+    @Override
+    public ScheduleObject getTaskNodeAlignment(
+            final List<Task> unscheduledTasks,
+            final Map<NodeWithAlloc, Requirements> availableByNode
+    ){
+        long start = System.currentTimeMillis();
+        if ( traceEnabled ) {
+            int index = 1;
+            for ( Task unscheduledTask : unscheduledTasks ) {
+                unscheduledTask.getTraceRecord().setSchedulerPlaceInQueue( index++ );
+            }
+        }
+        prioritize.sortTasks( unscheduledTasks );
+        
+
+        unscheduledTasks.stream().map(obj -> obj.getConfig().getName()).forEach(System.out::println);
+
+        // first alignemnt (LabelAssign)
+        List<NodeTaskAlignment> alignmentLabelAssign = nodeLabelAssigner.getTaskNodeAlignment(unscheduledTasks, availableByNode);
+        List<String> namesList = alignmentLabelAssign.stream().map(obj -> obj.task.getConfig().getName()).collect(Collectors.toList());
```

Comment:
listOfUnassignedTasks

Reply:
I think it is better to make variable alignmentLabelAssign from type NodeLabelAssign and then store a list of unscheduled tasks and fetch it here. Then you don't need to calculate this here again.